modules/services/way-displays.nix

```diff
@@ -77,35 +95,49 @@ in
   config = mkIf cfg.enable {
     assertions = [
       (lib.hm.assertions.assertPlatform "services.way-displays" pkgs lib.platforms.linux)
+      {
+        assertion = cfg.statefulConfiguration -> cfg.settings == null;
+        message = "way-displays: when `statefulConfiguration = true` then `settings` must be `null`";
+      }
+      {
+        assertion = (!cfg.statefulConfiguration) -> cfg.settings != null;
+        message = "way-displays: `settings` must be non-null unless `statefulConfiguration = true`";
+      }
     ];
 
-    xdg.configFile."way-displays/cfg.yaml".source =
-      yaml.generate "way-displays-config.yaml"
-        (mergeSets [
-          {
-            CALLBACK_CMD = "${pkgs.libnotify}/bin/notify-send \"way-displays \${CALLBACK_LEVEL}\" \"\${CALLBACK_MSG}\"";
-          }
-          cfg.settings
-        ]);
+    home.packages = [ cfg.package ];
 
-    systemd.user.services.way-displays = {
-      Unit = {
-        Description = "Display configuration service";
-        Documentation = "man:way-displays(1)";
-        ConditionEnvironment = "WAYLAND_DISPLAY";
-        PartOf = cfg.systemdTarget;
-        Requires = cfg.systemdTarget;
-        After = cfg.systemdTarget;
-      };
+    xdg.configFile."way-displays/cfg.yaml" = mkIf (!cfg.statefulConfiguration) {
+      source = yaml.generate "way-displays-config.yaml" (mergeSets [
+        {
+          CALLBACK_CMD = "${pkgs.libnotify}/bin/notify-send \"way-displays \${CALLBACK_LEVEL}\" \"\${CALLBACK_MSG}\"";
+        }
+        cfg.settings
+      ]);
+    };
+
+    systemd.user.services.${serviceName} = {
+      Unit =
+        {
+          Description = "Display configuration service";
+          Documentation = "man:way-displays(1)";
+          ConditionEnvironment = "WAYLAND_DISPLAY";
+          Requisite = cfg.systemdTarget;
```

Comment:
`Requires` -> `Requisite`

I think that's more appropriate anyway because with `Requires` starting way-displays would trigger the dependency to start. That doesn't seem expected to me.

Reply:
Now out of scope.